configure.ac

```diff
@@ -2045,7 +2046,8 @@ then
   ac_cv_cxx_thread=yes
 elif test "$ac_cv_pthread" = "yes"
 then
-  CXX="$CXX -pthread"
+  CXX="$CXX"
+  CXXFLAGS="-pthread"
```

Comment:
this also stamps over whatever `CXXFLAGS` was before

Also, it seems the cases above need this treatment?